### PR TITLE
Use an SQLAlchemy relationship between annotation and group instead of the service

### PR DESCRIPTION
**Pull request by jon-betts**
For: https://github.com/hypothesis/h/issues/6769

This PR:

 * Introduces a relationship to go from annotations to groups
 * Uses it one view to prove it works

### CODE REVIEW

File: h/storage.py [hunk 1]
```diff
@@ -184,7 +176,7 @@ def update_annotation(request, id_, data, group_service):
     annotation = request.db.query(models.Annotation).get(id_)
     annotation.updated = updated
 
-    group = group_service.find(annotation.groupid)
+    group = request.find_service(IGroupService).find(annotation.groupid)
```
Comment (marcospri) [inline h/storage.py:179]: annotation.group here too?

Comment (jon-betts) [reply h/storage.py]: Yeah unfortunately this is was the source of loads of pain

File: tests/h/views/api/flags_test.py [hunk 2]
```diff
@@ -110,37 +62,25 @@ def test_doesnt_send_email_if_group_creator_has_no_email_address(
...
-            return_value=(
-                ["fake@example.com"],
-                "Some subject",
-                "Some text",
-                "Some html",
-            ),
-        )
+    @pytest.fixture(autouse=True)
+    def flag_notification(self, patch):
+        return patch("h.views.api.flags.flag_notification")
```
Comment (jon-betts) [inline tests/h/views/api/flags_test.py:78]: This was patching at source not destination

File: tests/h/views/api/flags_test.py [hunk 3]
```diff
@@ -110,37 +62,25 @@ def test_doesnt_send_email_if_group_creator_has_no_email_address(
...
+            instance=True,
+            annotation=factories.Annotation(group=factories.Group()),
         )
 
     @pytest.fixture
-    def pyramid_request(self, factories, pyramid_request):
+    def pyramid_request(self, factories, pyramid_request, annotation_context):
         pyramid_request.user = factories.User()
-        pyramid_request.json_body = {}
+        pyramid_request.json_body = {"annotation": annotation_context.annotation.id}
```
Comment (jon-betts) [inline tests/h/views/api/flags_test.py:73]: Make the default a call and deviate from there

**Review by marcospri — APPROVED** [review]

**Pull request merged**